test/e2e/framework/job/wait.go

```diff
@@ -60,19 +59,38 @@ func WaitForJobPodsSucceeded(ctx context.Context, c clientset.Interface, ns, job
 
 // waitForJobPodsInPhase wait for all pods for the Job named JobName in namespace ns to be in a given phase.
 func waitForJobPodsInPhase(ctx context.Context, c clientset.Interface, ns, jobName string, expectedCount int32, phase v1.PodPhase, timeout time.Duration) error {
-	return wait.PollUntilContextTimeout(ctx, framework.Poll, timeout, false, func(ctx context.Context) (bool, error) {
-		pods, err := GetJobPods(ctx, c, ns, jobName)
-		if err != nil {
-			return false, err
-		}
-		count := int32(0)
-		for _, p := range pods.Items {
-			if p.Status.Phase == phase {
-				count++
+	return framework.Gomega().
+		Eventually(ctx, func(ctx context.Context) (*v1.PodList, error) {
+			return GetJobPods(ctx, c, ns, jobName)
+		}).
+		WithPolling(framework.Poll).
+		WithTimeout(timeout).
+		Should(framework.MakeMatcher(func(pods *v1.PodList) (func() string, error) {
+			var nonMatchingPods, matchingPods []v1.Pod
+			for _, p := range pods.Items {
+				if p.Status.Phase == phase {
+					matchingPods = append(matchingPods, p)
+				} else {
+					nonMatchingPods = append(nonMatchingPods, p)
+				}
 			}
-		}
-		return count == expectedCount, nil
-	})
+
+			if len(matchingPods) > int(expectedCount) {
+				return func() string {
+					return fmt.Sprintf("expected at most %d pods in phase %s, %d out of %d were not:\n%s",
```

Comment:
Please log the job name here using `klog.KRef` to keep consistency

Reply:
will do